pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -834,21 +833,10 @@ func LoadIPSecKeys(r io.Reader) (int, uint8, error) {
 			return 0, 0, fmt.Errorf("missing IPSec key or invalid format")
 		}
 
-		spiI, err := strconv.Atoi(s[offsetSPI])
+		spi, offsetBase, err = parseSPI(s[offsetSPI])
 		if err != nil {
-			// If no version info is provided assume using key format without
-			// versioning and assign SPI.
-			log.Warning("IPsec secrets without an SPI as the first argument are deprecated and will be unsupported in v1.13.")
-			spiI = 1
-			offsetBase = -1
+			return 0, 0, err
```

Comment:
Nit:
```suggestion
			return 0, 0, fmt.Errorf("parsing SPI: %w, err)
```
To group the possible errors that are returned from the function?